src/graphapi/engine.cpp

```diff
@@ -32,52 +32,111 @@ namespace miopen {
 
 namespace graphapi {
 
-EngineBuilder& EngineBuilder::setOpGraph(const OpGraph* opGraph)
+GraphPatternExecutor::~GraphPatternExecutor() = default;
+
+size_t GraphExecutorFind20::getWorkspaceSize() const
 {
-    mOpGraph = checkPtr(opGraph);
-    return *this;
+    size_t workspace_size = 0;
+    auto s                = miopenGetSolutionWorkspaceSize(mSolution, &workspace_size);
+    MIOPEN_THROW_IF(s != miopenStatusSuccess, "get solution workspace size failed");
+
+    return workspace_size;
 }
 
-EngineBuilder& EngineBuilder::setGlobalIndex(int64_t globalIndex)
+void GraphExecutorFind20::execute(miopenHandle_t handle, const VariantPack& vpk)
 {
-    if(globalIndex >= 0)
+
+    std::vector<miopenTensorArgument_t> tens_args;
+
+    auto num = vpk.getTensorIds().size();
+    assert(num == vpk.getDataPtrs().size());
+
+    // TODO(amber) : verify that variant pack has all the expected input and output
+    // tensors
+    for(auto i = 0ull; i < num; ++i)
```

Comment:
[R] You do not need to derive the type of variable from `auto... = ...ull` ;)
```suggestion
    for(std::size_t i = 0; i < num; ++i)
```